### PR TITLE
Add support for k8s 1.15

### PR DESCRIPTION
**Pull request by aanm**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8280)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-docs-please

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Comment by aanm** [comment]
test-upstream-k8s

**Review by raybejjani — APPROVED** [review]

**Comment by aanm** [comment]
test-upstream-k8s

**Comment by aanm** [comment]
test-upstream-k8s

**Comment by aanm** [comment]
test-me-please

File: Documentation/contributing.rst [hunk 1]
```diff
@@ -1381,17 +1382,22 @@ Cilium-PR-Ginkgo-Tests-k8s
 Runs the Kubernetes e2e tests against all Kubernetes versions that are not
 currently not tested as part of each pull-request, but which Cilium still
 supports, as well as the the most-recently-released versions of Kubernetes that
-are not yet declared stable by Kubernetes upstream:
+are might not be declared stable by Kubernetes upstream:
```
Comment (ianvernon) [inline Documentation/contributing.rst]: Nit: "that might not be declared"

File: Documentation/contributing.rst [hunk 2]
```diff
@@ -1381,17 +1382,22 @@ Cilium-PR-Ginkgo-Tests-k8s
...
 
-    - 1.9
-    - 1.10
+    - 1.12
+    - 1.13
+
+Third stage
+
+    - 1.14
+    - beta versions (1.16-beta once its out)
```
Comment (ianvernon) [inline Documentation/contributing.rst]: Nit: it's

File: kubernetes-upstream.Jenkinsfile [hunk 3]
```diff
@@ -74,6 +74,15 @@ pipeline {
             steps {
                 sh 'cd ${TESTDIR}; vagrant up k8s1-${K8S_VERSION}'
                 sh 'cd ${TESTDIR}; vagrant up k8s2-${K8S_VERSION}'
+            }
+        }
+
+        stage('BDD-tests'){
```
Comment (ianvernon) [inline kubernetes-upstream.Jenkinsfile:80]: Thanks for adding this, hopefully this will help with the provisioning timeout problems on this job.

**Review by ianvernon — APPROVED** [review]
Nothing blocking - feel free to address the documentation nits after all the tests hav passed so that way you can just run the documentation build without having to run other builds too.

**Comment by aanm** [comment]
all tests have previously passed. running only documentation CI to address Ian's comments

**Comment by aanm** [comment]
test-docs-please

**Comment by aanm** [comment]
merging, all tests have passed in previous runs.

**Pull request merged**